sympy/solvers/inequalities.py

```diff
@@ -395,109 +378,117 @@ def solve_univariate_inequality(expr, gen, relational=True):
-        sol_sets = [S.EmptySet]
-        try:
-            reals = _nsort(set(solns + singularities), separated=True)[0]
-        except NotImplementedError:
-            raise NotImplementedError('sorting of these roots is not supported')
-        for x in reals:
-            end = x
-
-            if end in [S.NegativeInfinity, S.Infinity]:
-                if valid(S(0)):
-                    sol_sets.append(Interval(start, S.Infinity, True, True))
-                    break
-
-            if valid((start + end)/2 if start != S.NegativeInfinity else end - 1):
+                reals = _nsort(set(solns + singularities), separated=True)[0]
+            except NotImplementedError:
+                raise NotImplementedError('sorting of these roots is not supported')
+            for x in reals:
+                end = x
+
+                if end in [S.NegativeInfinity, S.Infinity]:
+                    if valid(S(0)):
+                        sol_sets.append(Interval(start, S.Infinity, True, True))
+                        break
+
+                if valid((start + end)/2 if start != S.NegativeInfinity else end - 1):
+                    sol_sets.append(Interval(start, end, True, True))
+
+                if x in singularities:
+                    singularities.remove(x)
+                elif include_x:
+                    sol_sets.append(FiniteSet(x))
+
+                start = end
+
+            end = S.Infinity
+
+            # in case start == -oo then there were no solutions so we just
+            # check a point between -oo and oo (e.g. 0) else pick a point
+            # past the last solution (which is start after the end of the
+            # for-loop above
+            if valid(start + 1 if start is not S.NegativeInfinity else 0):
                 sol_sets.append(Interval(start, end, True, True))
 
-            if x in singularities:
-                singularities.remove(x)
-            elif include_x:
-                sol_sets.append(FiniteSet(x))
-
-            start = end
+            rv = Union(*sol_sets)
 
-        end = S.Infinity
+    return rv if not relational else rv.as_relational(gen)
 
-        # in case start == -oo then there were no solutions so we just
-        # check a point between -oo and oo (e.g. 0) else pick a point
-        # past the last solution (which is start after the end of the
-        # for-loop above
-        if valid(start + 1 if start is not S.NegativeInfinity else 0):
-            sol_sets.append(Interval(start, end, True, True))
 
-        rv = Union(*sol_sets).subs(gen, _gen)
-
-    return rv if not relational else rv.as_relational(_gen)
-
-
-def _solve_inequality(ie, s):
-    """ A hacky replacement for solve, since the latter only works for
-        univariate inequalities. """
-    expr = ie.lhs - ie.rhs
-    try:
-        p = Poly(expr, s)
-        if p.degree() != 1:
-            raise NotImplementedError
-    except (PolynomialError, NotImplementedError):
+def solve_linear_inequality(ie, s):
```

Comment:
On second thought it won't hurt to leave it private until it is really clear how the various functions for solving/handling inequalities should work.